src/pocket/PocketSQSWithLambdaTarget.ts

```diff
@@ -15,26 +15,38 @@ import {
   IamRole,
   IamRolePolicyAttachment,
   LambdaEventSourceMapping,
+  DataAwsSqsQueue,
+  DataAwsSqsQueueConfig,
+  SqsQueue,
 } from '../../.gen/providers/aws';
 
 export interface PocketSQSWithLambdaTargetProps
   extends PocketVersionedLambdaProps {
-  sqsQueue?: {
-    messageRetentionSeconds?: number;
-    maxReceiveCount?: number;
-    maxMessageSize?: number;
-    delaySeconds?: number;
-    visibilityTimeoutSeconds?: number;
-  };
+  /**
+   * Set dataSqsQueue to use an existing SQS queue. If not provided, then a queue will be created.
+   */
+  configFromPreexistingSqsQueue?: DataAwsSqsQueueConfig;
+  /**
+   * Configure a new SQS queue. Cannot be used in combination with configFromPreexistingSqsQueue.
+   */
+  sqsQueue?: PocketSQSProps;
   batchSize?: number;
   batchWindow?: number;
 }
 
+export interface PocketSQSProps {
+  messageRetentionSeconds?: number;
+  maxReceiveCount?: number;
+  maxMessageSize?: number;
+  delaySeconds?: number;
+  visibilityTimeoutSeconds?: number;
+}
+
 /**
  * Extends the base pocket versioned lambda class to add a sqs based trigger on top of the lambda
  */
 export class PocketSQSWithLambdaTarget extends PocketVersionedLambda {
-  public readonly sqsQueue: ApplicationSQSQueue;
+  private readonly sqsQueueResource: SqsQueue | DataAwsSqsQueue;
```

Comment:
This will create a new version of the package. Even if there are existing usages that I haven't found any, they won't be impacted until updated.